src/main/java/io/jenkins/plugins/sample/OnboardingSectionConfiguration.java

```diff
@@ -0,0 +1,245 @@
+        this.categoryConfig = categoryConfig;
+    }
+
+    public FormValidation doCheckName(@QueryParameter String value) {
+        if(Objects.isNull(value) || value.isBlank()) {
+            return FormValidation.warning("Name is empty");
+        } else {
+            if (!validateName(value)) {
+                return FormValidation.warning("Invalid name data");
+            }
+        }
+        return FormValidation.ok("All good!");
+    }
+
+    private boolean validateName(String name) {
+        Pattern pattern = Pattern.compile("^[A-Za-z\\- ]+$");
+        if (!pattern.matcher(name).matches()) {
+            return false;
+        }
+        return true;
+    }
+
+    public FormValidation doCheckUsername(@QueryParameter String value) {
+        if(Objects.isNull(value) || value.isBlank()) {
+            return FormValidation.warning("Username is empty");
+        } else {
+            Pattern pattern = Pattern.compile("^[A-Za-z]+$");
+            if (!pattern.matcher(value).matches()) {
+                return FormValidation.warning("Invalid Username Input");
+            }
+        }
+        return FormValidation.ok("All good!");
+    }
+
+    public FormValidation doCheckPassword(@QueryParameter String value) {
+        if(Objects.isNull(value) || value.isBlank()) {
+            return FormValidation.warning("Password is empty");
+        } else {
+        }
+        return FormValidation.ok("All good!");
+    }
+
+    @POST
+    public FormValidation doTestConnection(@QueryParameter("username") final String username,
+                                           @QueryParameter("password") final String password,
+                                           @QueryParameter("url") String url,
+                                           @AncestorInPath Job job) {
+        if (url.isEmpty()) {
+            return FormValidation.error("URL cannot be empty");
+        } else if (!url.startsWith("http")) {
+            return FormValidation.error("Enter URL in correct format");
+        }
+        if (username.isEmpty() || password.isEmpty()) {
+            return FormValidation.error("Enter username and password.");
+        }
+        try {
+            if (job == null) {
+                Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+            } else {
+                job.checkPermission(Item.CONFIGURE);
+            }
+
+            return invokeRestApiPost(url, null);
+        } catch (Exception e) {
+            return FormValidation.error("Client error : "+ e.getMessage());
+        }
+    }
+
+    private static FormValidation invokeRestApiPost(String url, String authHeader) {
+        return invokeRestApiPost(url, authHeader, null);
+    }
+    private static FormValidation invokeRestApiPost(String url, String authHeader, String payload) {
+        try {
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest.Builder requestBuilder = HttpRequest.newBuilder()
```

Comment:
FTR in Jenkins use [these methods](https://github.com/jenkinsci/jenkins/blob/b0e47a7a9bf5b448bb906e7087cac841b2b5b144/core/src/main/java/hudson/ProxyConfiguration.java#L362-L432) to create `HttpClient` instances to respect users' proxy configurations.